### PR TITLE
azure/persist: fallback to copy if rename fails

### PR DESCRIPTION
**Pull request by colemickens**
cc: @ahmetalpbalkan

### CODE REVIEW

**Comment by ahmetalpbalkan** [comment]
hmm don't you think fallback logic just makes it harder to maintain/trace? If our tmpdir was `dirname(dest)` in the first place, we wouldn't even bother working with /tmp.

**Comment by ahmetalpbalkan** [comment]
what I was trying to say is, this logic seems a lot complicated, if we can just do `os.TempFile(dirname(dest), "...")`, write the new file there and do Rename it would just work. I think the main issue was copying across filesystems, no? If we create a temp file next to the destination file, we won't have this problem?

**Comment by colemickens** [comment]
I follow. I agree. Will change.

**Comment by colemickens** [comment]
Changed per your feedback. Tested with the example in the repo and `-tokenCachePath="$HOME/temptmpfs/cache/token.json"` and `$HOME/temptmpfs` is a directory with `tmpfs` mounted in, so it's a different FS.

**Comment by colemickens** [comment]
(I'll squash these, leaving them separate commits for reviewing purposes)

File: autorest/azure/persist.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func SaveToken(path string, mode os.FileMode, token Token) error {
 
 	// Atomic replace to avoid multi-writer file corruptions
 	if err := os.Rename(tempPath, path); err != nil {
-		return fmt.Errorf("failed to move temporary token to desired output location. source=(%s). destination=(%s). error = %v", tempPath, path, err)
+		return fmt.Errorf("failed to move temporary token to desired output location. source=(%s). destination=(%s). error(rename)=%q", tempPath, path, err)
```
Comment (ahmetalpbalkan) [inline autorest/azure/persist.go]: the format can be better as:

    failed to move temporary token to desired output location: src=%q dsts=%q: %v

**Comment by ahmetalpbalkan** [comment]
LGTM

**Comment by ahmetalpbalkan** [comment]
thanks

**Pull request merged**